bundle/src/main/java/com/adobe/acs/commons/remoteassets/impl/RemoteAssetsNodeSyncImpl.java

```diff
@@ -0,0 +1,609 @@
+     * Grab JSON from the provided path and sync to the JCR.
+     * @param nextPath String
+     * @param primaryType String
+     * @return Node
+     */
+    private Node getOrCreateNode(final String nextPath, final String primaryType) {
+        Node node = null;
+
+        try {
+            node = JcrUtils.getNodeIfExists(nextPath, this.session);
+            if (node == null) {
+                node = JcrUtil.createPath(nextPath, primaryType, this.session);
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("New node '{}' created.", node.getPath());
+                }
+            } else {
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("Node '{}' retrieved from JCR.", node.getPath());
+                }
+            }
+        } catch (RepositoryException e) {
+            LOG.error("Repository Exception. Unable to get or create node '{}'", nextPath, e);
+        }
+
+        return node;
+    }
+
+    /**
+     * Get {@link JSONObject} from URL response.
+     * @param path String
+     * @return JSONObject
+     * @throws IOException exception
+     * @throws JSONException exception
+     */
+    private JSONObject getJsonFromUri(final String path) throws IOException, JSONException {
+        try {
+            URI pathUi = new URI(null, null, path, null);
+            // we want to traverse the JCR one level at a time, hence the '1' selector.
+            URL url = new URL(this.remoteAssetsConfig.getServer().concat(pathUi.toString()).concat(".1.json"));
+            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+            String rawAuth = String.format("%s:%s", this.remoteAssetsConfig.getUsername(), this.remoteAssetsConfig.getPassword());
+            String encodedAuth = Base64.getEncoder().encodeToString(rawAuth.getBytes(StandardCharsets.UTF_8));
+            connection.setRequestProperty("Authorization", String.format("Basic %s", encodedAuth));
+
+            InputStream is = connection.getInputStream();
+            InputStreamReader isReader = new InputStreamReader(is);
+            BufferedReader br = new BufferedReader(isReader);
+            StringBuilder sb = new StringBuilder();
+
+            try {
+                String line;
+                while ((line = br.readLine()) != null) {
+                    sb.append(line);
+                }
+
+                String sbString = sb.toString();
+
+                if (StringUtils.startsWith(sbString, "{")) {
+                    return new JSONObject(sbString);
+                } else {
+                    LOG.error("Unable to grab JSON Object. Please ensure URL {} is valid. \nRaw Response: {}", url.toString(), sbString);
+                }
+            } finally {
+                is.close();
+                isReader.close();
+                br.close();
+
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("JSON successfully fetched for URL '{}'.", url.toString());
```

Comment:
Since we're in a `finally` block I dont think we know if the JSON was successfully fetched or not.  Might make more sense to instead put a statement similar to this but right before we try to call the URL.